### PR TITLE
[PUBDEV-7029] add support for predictions with offset/base_margin

### PR DESCRIPTION
**Pull request by honzasterba**

### CODE REVIEW

File: xgboost-predictor/src/main/java/biz/k11i/xgboost/Predictor.java [hunk 1]
```diff
@@ -153,17 +178,17 @@ void initObjGbm() {
...
             preds = obj.predTransform(preds);
         }
         return preds;
     }
 
-    float[] predictRaw(FVec feat, int ntree_limit) {
+    float[] predictRaw(FVec feat, float base_score, int ntree_limit) {
         float[] preds = gbm.predict(feat, ntree_limit);
         for (int i = 0; i < preds.length; i++) {
-            preds[i] += mparam.base_score;
```
Comment (michalkurka) [inline xgboost-predictor/src/main/java/biz/k11i/xgboost/Predictor.java]: this seems incorrect to me, is there any other place where the base score would be applied (base_score IMHO corresponds to H2O's GBM `init_f`)?

File: xgboost-predictor/src/main/java/biz/k11i/xgboost/Predictor.java [hunk 2]
```diff
@@ -144,6 +144,31 @@ void initObjGbm() {
...
+    /**
+     * Generates predictions for given feature vector.
+     *
+     * @param feat          feature vector
+     * @param base_margin   predict with base margin for each prediction
+     * @param ntree_limit   limit the number of trees used in prediction
+     * @return prediction values
+     */
+    public float[] predict(FVec feat, float base_margin, int ntree_limit) {
+        float[] preds = predictRaw(feat, base_margin, ntree_limit);
```
Comment (michalkurka) [inline xgboost-predictor/src/main/java/biz/k11i/xgboost/Predictor.java:167]: hmm... do the models that are trained with base_margin necessarily `mparam.base_score == 0`, this is not obvious to me

Comment (honzasterba) [reply xgboost-predictor/src/main/java/biz/k11i/xgboost/Predictor.java:167]: based on the original xgboost code it seems to me that with base_margin used base_score is ignored, will verify with tests

Comment (michalkurka) [reply xgboost-predictor/src/main/java/biz/k11i/xgboost/Predictor.java:167]: ok, thanks @honzasterba

**Pull request merged**